tests/publiccloud/registration.pm

```diff
@@ -27,6 +27,11 @@ sub run {
 
     registercloudguest($args->{my_instance}) if (is_byos() || get_var('PUBLIC_CLOUD_FORCE_REGISTRATION'));
     register_addons_in_pc($args->{my_instance});
+    # Double confirm system is correctly registered, and quit earlier if anything wrong
+    # see bsc#1253777, we may need have to rerun the failed job in this case
+    record_info('Check registration status');
+    my $reg_status = $args->{my_instance}->ssh_assert_script_run("sudo SUSEConnect -s");
+    die "System is not correctly registered" if ($reg_status =~ /Not Registered/m);
```

Comment:
This is a very good idea, but we're right now in the process of collecting more information about the registration issues. For this we need the output of `zypper`.

So, this change is really good, but let's merge this only after the data collection has been completed.

@asmorodskyi please take a look

Reply:
Thanks all!